### ISSUE TITLE
Issues with tearoffs

### ISSUE THREAD

**Issue by zerny** [dart-lang/fletch#19]
Currently tearoffs are not considered functions and can not be called without disambiguating the receiver at the call site:

```dart
class Foo {
  Function fn;
  Foo() {
    fn = () => 42;
  }
}

main() {
  Foo foo = new Foo();
  print(foo.fn is Function); // should print true, currently prints false.
  print((foo.fn)()); // prints 42.
  print(foo.fn()); // results in an error.
}
```

**Comment by skabet** [comment]
Proposed fix: https://codereview.chromium.org/1103663002/

**Comment by kasperl** [comment]
@skabet The patch only solves the 'is Function' part right?

**Comment by skabet** [comment]
Ah yes, I meant to split this up. There are two issues here.

**Comment by skabet** [comment]
See #20 for the closure in field problem.

**Comment by skabet** [comment]
Closing as eb63ad274f0d19aa953d9426dec4e816ec439557 landed, and #20 is tracing the field problem.
